pkg/runtime/runtime.go

```diff
@@ -2162,6 +2178,27 @@ func (a *DaprRuntime) isComponentAuthorized(component componentsV1alpha1.Compone
 	return true
 }
 
+func (a *DaprRuntime) IsComponentLoaded(component componentsV1alpha1.Component) bool {
+	a.componentsLock.RLock()
+	defer a.componentsLock.RUnlock()
+	for _, loadedComp := range a.components {
+		if strings.Compare(component.ObjectMeta.Name, loadedComp.ObjectMeta.Name) == 0 {
+			if strings.Compare(component.Spec.Type, loadedComp.Spec.Type) == 0 {
+				if strings.Compare(component.Spec.Version, loadedComp.Spec.Version) == 0 {
+					log.Warnf("Cannot load multiple instances of same component with dynamic loading. Component name already in use - name: %s type: %s/%s. Skipping dynamic loading.", component.ObjectMeta.Name, component.Spec.Type, component.Spec.Version)
```

Comment:
Actually we can load multiple instances of a component - say kafka. The challenge is using the same identifier. Also, this method only validates if a component matches an already loaded component. So there should not be any warnings logged in this method. IMHO it should be in the calling method.

Reply:
Changed to using reflect.DeepEqual() for exact component match. Warning now logged from calling method.